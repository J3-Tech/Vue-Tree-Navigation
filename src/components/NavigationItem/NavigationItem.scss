.NavigationItem {
  display:        inline-block;
  padding-top:    5px;
  padding-bottom: 5px;

  span {
    cursor: pointer;
  }
}
