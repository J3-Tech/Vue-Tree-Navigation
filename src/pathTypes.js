export default {
  PATH_TYPE_NONE: 'none',
  PATH_TYPE_ELEMENT: 'element',
  PATH_TYPE_ROUTE: 'route',
};
