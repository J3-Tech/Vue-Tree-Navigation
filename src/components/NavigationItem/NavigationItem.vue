/**
 * Return a navigation item - a hyperlink, a router link, or a text.
 * Return only item name in case it is neither the route nor the element.
 * Return a router link in case the item is route.
 * Return a hyperlink in case the item is element.
 */

<template>
  <div class="NavigationItem">
    <span v-if="showText">{{ item.name }}</span>

    <a
      v-if="showHyperLink"
      :href="item.meta.path">{{ item.name }}</a>

    <router-link
      v-if="showRouterLink"
      class="router-link"
      :to="item.meta.path">{{ item.name }}</router-link>
  </div>
</template>

<script>
import pathTypes from '../../pathTypes';
const { PATH_TYPE_NONE, PATH_TYPE_ELEMENT, PATH_TYPE_ROUTE } = pathTypes;

export default {
  props: {
    item: Object,
    required: true,
  },
  computed: {
    showText() {
      return this.item.meta.pathType === PATH_TYPE_NONE;
    },
    showHyperLink() {
      return this.item.meta.pathType === PATH_TYPE_ELEMENT;
    },
    showRouterLink() {
      return this.item.meta.pathType === PATH_TYPE_ROUTE;
    },
  },
};
</script>

<style lang="scss">
@import './NavigationItem.scss';
</style>
