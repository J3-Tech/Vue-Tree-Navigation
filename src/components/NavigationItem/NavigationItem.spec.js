import { shallow } from 'vue-test-utils';

import pathTypes from '../../pathTypes';
import NavigationItem from './NavigationItem';

const { PATH_TYPE_NONE, PATH_TYPE_ELEMENT, PATH_TYPE_ROUTE } = pathTypes;

const textItem = {
  name: 'Contact',
  meta: {
    pathType: PATH_TYPE_NONE,
  },
};

const elementItem = {
  name: 'Contact',
  element: 'contact',
  meta: {
    pathType: PATH_TYPE_ELEMENT,
    path: '#contact',
  },
};

const routeItem = {
  name: 'About',
  route: 'about',
  meta: {
    pathType: PATH_TYPE_ROUTE,
    path: '/about',
  },
};

describe('NavigationItem ', () => {
  context('for item with path type none', () => {
    it('renders span containing the item name', () => {
      const wrapper = shallow(NavigationItem, {
        propsData: {
          item: textItem,
        },
      });

      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  context('for item with path type element', () => {
    it('renders hyperlink with a correct value and location', () => {
      const wrapper = shallow(NavigationItem, {
        propsData: {
          item: elementItem,
        },
      });

      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  context('for item with path type route', () => {
    it('renders router link with a correct value and location', () => {
      const wrapper = shallow(NavigationItem, {
        propsData: {
          item: routeItem,
        },
      });

      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
