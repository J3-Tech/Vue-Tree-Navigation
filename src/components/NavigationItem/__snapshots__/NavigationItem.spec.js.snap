// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItem  for item with path type element renders hyperlink with a correct value and location 1`] = `
<div class="NavigationItem">
  <!---->
  <a href="#contact">Contact</a>
  <!---->
</div>
`;

exports[`NavigationItem  for item with path type none renders span containing the item name 1`] = `
<div class="NavigationItem"><span>Contact</span>
  <!---->
  <!---->
</div>
`;

exports[`NavigationItem  for item with path type route renders router link with a correct value and location 1`] = `
<div class="NavigationItem">
  <!---->
  <!---->
  <router-link to="/about" class="router-link">About</router-link>
</div>
`;
